client/state/preferences/reducer.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import get from 'lodash/get';
+import mapValues from 'lodash/mapValues';
+import { createReducer } from 'state/utils';
+/**
+ * Internal dependencies
+ */
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_FETCH,
+	PREFERENCES_FETCH_SUCCESS,
+	PREFERENCES_FETCH_FAILURE,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+import { USER_SETTING_KEY, DEFAULT_PREFERENCES } from './constants';
+
+export function createReducerForPreferenceKey( key, defaultValue = null ) {
+	const handlePreferencesFromApi = ( state, action ) => get( action, [ 'data', USER_SETTING_KEY, key ], state );
+	return createReducer( defaultValue, {
+		[ PREFERENCES_SET ]: ( state, action ) => ( action.key === key ) ? action.value : state,
+		[ PREFERENCES_RECEIVE ]: handlePreferencesFromApi,
+		[ PREFERENCES_FETCH_SUCCESS ]: handlePreferencesFromApi,
+		[ SERIALIZE ]: state => state,
+		[ DESERIALIZE ]: state => state,
```

Comment:
Ah, well in this case we add a json schema to avoid [data drift issues](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#solution-schema-validation). Do we have a static list of preference keys, or do we store arbitrary key, value pairs?